File: Promotion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

/**
 *
 * @author Med Aziz
 */
public class Promotion {
    
    int id;
    String nom,type,description,pourcentage,categorie;
    private String code;
public static int pdp ;

    public static int getPdp() {
        return pdp;
    }

    public static void setPdp(int pdp) {
        Promotion.pdp = pdp;
    }


    public Promotion() {
    }

    public Promotion(int id, String nom, String type, String description, String pourcentage,String categorie,String code) {
        this.id = id;
        this.nom = nom;
        this.type = type;
        this.description = description;
        this.pourcentage = pourcentage;
        this.categorie = categorie;
        this.code=code;

    }
    

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getCategorie() {
        return categorie;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }
    

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPourcentage() {
        return pourcentage;
    }

    public void setPourcentage(String pourcentage) {
        this.pourcentage = pourcentage;
    }

    @Override
    public String toString() {
        return "Promotion{" + "id=" + id + ", nom=" + nom + ", type=" + type + ", description=" + description + ", pourcentage=" + pourcentage + ", categorie=" + categorie + '}';
    }

    
    
      
    
}

File: Promotion.php
<?php

namespace App\Entity;

use App\Entity\Film;
use App\Repository\PromotionRepository;
use Doctrine\ORM\Mapping as ORM;
use  Symfony\Component\Validator\Constraints as Assert;

use App\Form\SearchPromotionType;

/**
 * @ORM\Entity(repositoryClass=PromotionRepository::class)
 * @ORM\Table(name="promotion", indexes={@ORM\Index(columns={"type","description"},flags={"fulltext"})})
 */
class Promotion
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $nom;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $type;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $description;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $pourcentage;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $code;

    /**
     * @ORM\ManyToOne(targetEntity=Film::class)
     * @ORM\JoinColumn(nullable=false)
     */
    private $cat;



    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getPourcentage(): ?string
    {
        return $this->pourcentage;
    }

    public function setPourcentage(string $pourcentage): self
    {
        $this->pourcentage = $pourcentage;

        return $this;
    }

    public function getCode(): ?string
    {
        return $this->code;
    }

    public function setCode(string $code): self
    {
        $this->code = $code;

        return $this;
    }


  public function getCat(): ?Film
  {
      return $this->cat;
  }

  public function setCat(?Film $cat): self
  {
      $this->cat = $cat;

      return $this;
  }

      public function __toString()
      {
          return (string) $this->getCat();

      }

}
